### PR TITLE
Cassandra pilot

### PR DESCRIPTION
**Pull request by wallrj**
* A pilot command for managing a Cassandra subprocess
* Installed into the Cassandra container via an InitContainer
* Watches for a Pilot resource with the name and namespace, which are supplied as command line arguments.
* Then launches the Cassandra subprocess.

Fixes: #141 

**Release note**:
```release-note
NONE
```

### CODE REVIEW

File: pkg/controllers/cassandra/nodepool/resource.go [hunk 1]
```diff
@@ -52,6 +52,48 @@ func StatefulSetForCluster(
...
+										Command: []string{
+											"/usr/bin/timeout",
+											"5",
+											"/ready-probe.sh",
+										},
+									},
+								},
+								InitialDelaySeconds: 60,
+								// XXX Kubernetes ignores the TimeoutSeconds for Exec probes.
+								// See https://github.com/kubernetes/kubernetes/issues/26895
```
Comment (munnerz) [inline pkg/controllers/cassandra/nodepool/resource.go]: Just FYI - as GenericPilot exposes HTTP health checks, we should be able to work around this once this becomes a Pilot probe 😄

File: hack/e2e.sh [hunk 2]
```diff
@@ -27,6 +27,10 @@ mkdir -p $TEST_DIR
 
 source "${SCRIPT_DIR}/libe2e.sh"
 
+# Mandatory environment variables
+: ${CHART_VALUES:?}
+: ${CHART_VALUES_CASSANDRA:?}
```
Comment (munnerz) [inline hack/e2e.sh]: Given we can now toggle which e2e tests to run (i.e. I could not run cassandra e2e), this var isn't *always* required. I can see it's defaulted in the Makefile itself, but would it make sense to move some of these defaults into this file? (perhaps including CHART_VALUES)?

File: hack/e2e.sh [hunk 3]
```diff
@@ -37,10 +41,6 @@ function debug_navigator_start() {
 }
 
 function helm_install() {
-    if [ "${CHART_VALUES}" == "" ]; then
-        echo "CHART_VALUES must be set";
-        exit 1
-    fi
```
Comment (munnerz) [inline hack/e2e.sh]: Is this no longer required?

File: pkg/controllers/cassandra/nodepool/resource.go [hunk 4]
```diff
@@ -123,6 +149,13 @@ func StatefulSetForCluster(
 									ContainerPort: util.DefaultCqlPort,
 								},
 							},
+							VolumeMounts: []apiv1.VolumeMount{
+								{
+									Name:      sharedVolumeName,
+									MountPath: sharedVolumeMountPath,
+									ReadOnly:  false,
```
Comment (munnerz) [inline pkg/controllers/cassandra/nodepool/resource.go]: (nit) can we make this ReadOnly: true? Although ES also has readOnly: false set, so can be a follow up/pending discussion

Comment (wallrj) [reply pkg/controllers/cassandra/nodepool/resource.go]: Good idea. Done. For ElasticSearch too.

File: hack/e2e.sh (hunk 3 above)
Comment (wallrj) [reply hack/e2e.sh]: Nope. The `: ${CHART_VALUES:?}` causes the script to exit with almost that exact message if the variable isn't set.

File: hack/e2e.sh (hunk 2 above)
Comment (wallrj) [reply hack/e2e.sh]: Good idea. Done.

**Pull request merged**